var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -46,68 +51,112 @@ class NetcdfC(CMakePackage, AutotoolsPackage):
+        when="@4.9.0:4.9.1~mpi+parallel-netcdf",
+    )
 
     variant("mpi", default=True, description="Enable parallel I/O for netcdf-4")
     variant("parallel-netcdf", default=False, description="Enable parallel I/O for classic files")
     variant("hdf4", default=False, description="Enable HDF4 support")
     variant("pic", default=True, description="Produce position-independent code (for shared libs)")
     variant("shared", default=True, description="Enable shared library")
     variant("dap", default=False, description="Enable DAP support")
+    variant("byterange", default=False, description="Enable byte-range I/O")
     variant("jna", default=False, description="Enable JNA support")
     variant("fsync", default=False, description="Enable fsync support")
-    variant("zstd", default=True, description="Enable ZStandard compression", when="@4.9.0:")
+    variant("nczarr_zip", default=False, description="Enable NCZarr zipfile format storage")
     variant("optimize", default=True, description="Enable -O2 for a more optimized lib")
 
-    # It's unclear if cdmremote can be enabled if '--enable-netcdf-4' is passed
-    # to the configure script. Since netcdf-4 support is mandatory we comment
-    # this variant out.
-    # variant('cdmremote', default=False,
-    #         description='Enable CDM Remote support')
+    variant("szip", default=True, description="Enable Szip compression plugin")
+    variant("blosc", default=True, description="Enable Blosc compression plugin")
+    variant("zstd", default=True, description="Enable Zstandard compression plugin")
+
+    with when("build_system=cmake"):
+        # Based on the versions required by the root CMakeLists.txt:
+        depends_on("cmake@2.8.12:", type="build", when="@4.3.3:4.3")
+        depends_on("cmake@2.8.11:", type="build", when="@4.4.0:")
+        depends_on("cmake@3.6.1:", type="build", when="@4.5.0:")
+        depends_on("cmake@3.12:", type="build", when="@4.9.0:")
+        # Starting version 4.9.1, nczarr_test/CMakeLists.txt relies on the FILE_PERMISSIONS feature
+        # of the configure_file command, which is only available starting CMake 3.20:
+        depends_on("cmake@3.20:", type="test", when="@4.9.1:")
 
-    # The patch for 4.7.0 touches configure.ac. See force_autoreconf below.
     with when("build_system=autotools"):
-        depends_on("autoconf", type="build", when="@4.7.0,main")
-        depends_on("automake", type="build", when="@4.7.0,main")
-        depends_on("libtool", type="build", when="@4.7.0,main")
-    # CMake system can use m4, but Windows does not yet support
-    depends_on("m4", type="build", when=sys.platform != "win32")
+        for __s in itertools.chain(["@main"], _force_autoreconf_when):
+            with when(__s):
+                depends_on("autoconf", type="build")
+                depends_on("automake", type="build")
+                depends_on("libtool", type="build")
+                depends_on("m4", type="build")
+        del __s
```

Comment:
This will bite us at some point, when the argument is stored as is by some object :worried: So far they are string and they get transformed to specs internally, so it's fine. I wouldn't try to encourage people to use this idiom.